CMakeLists.txt

```diff
@@ -805,28 +811,30 @@ option(HELICS_INSTALL_CEREAL "Install a copy of the cereal headers used for comp
 
 mark_as_advanced(HELICS_INSTALL_CEREAL)
 
-if (HELICS_INSTALL_CEREAL)
-install(
-    DIRECTORY ThirdParty/cereal
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/helics/external/optional
-    COMPONENT cereal
-)
-add_library(cereal INTERFACE)
-add_library(HELICS::cereal ALIAS cereal)
-target_include_directories(cereal
-    INTERFACE
-        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/ThirdParty>
-        $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include/helics/external/optional>
-)
-install(TARGETS cereal ${HELICS_EXPORT_COMMAND} COMPONENT cereal)
+if (BUILD_SHARED_LIBS OR NOT DISABLE_STATIC_LIB_INSTALL)
```

Comment:
I am not sure BUILD_SHARED_LIBS is the right variable to use here,  I thought there was some cases where I turned it off after enabling another variable to prevent it from being used in the submodules since some of them use it as well.

Reply:
I don't think there is an alternative that would work at the moment. How about a variable like `HELICS_INSTALL_CPP14_HEADERS` that gets set when `BUILD_SHARED_LIBS` is used to enable the CXX shared library in the main CMakeLists.txt?

Reply:
I think it is OK as is,  I remember running into some trouble with the on the jsoncpp library but I explicitly disabled inside the cmake for adding jsoncpp.

So I think you are using the appropriate variable for checking if the C++ shared library is enabled.